### PR TITLE
Sink: Support for HANA Dialect

### PR DESCRIPTION
**Pull request by santi81**
Support HANA Dialect for the JDBC Sink Connector

### CODE REVIEW

**Comment by santi81** [comment]
Signed the Contributor License Agreement

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    map.put(Schema.Type.BYTES, "BLOB");
+    return map;
+  }
+
+  @Override
+  public String getCreateQuery(String tableName, Collection<SinkRecordField> fields) {
+
+    // Defaulting to Column Stores
+    String createQuery = super.getCreateQuery(tableName, fields).replace("CREATE TABLE", "CREATE COLUMN TABLE");
+    log.trace("Create Query : " + createQuery);
```
Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java]: This is already logged at `INFO`  by the caller, so let's remove this log line

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  @Override
+  public List<String> getAlterTable(String tableName, Collection<SinkRecordField> fields) {
+    final StringBuilder builder = new StringBuilder("ALTER TABLE ");
+    builder.append(escaped(tableName));
+    builder.append(" ADD(");
+    writeColumnsSpec(builder, fields);
+    builder.append(")");
+    log.trace("Alter Table query: " + builder.toString());
```
Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java]: Also logged at INFO by the caller, so let's remove this log line

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    StringBuilder builder = new StringBuilder("UPSERT ");
+    builder.append(escaped(table));
+    builder.append("(");
+    joinToBuilder(builder, ",", keyCols, cols, escaper());
+    builder.append(") VALUES(");
+    nCopiesToBuilder(builder, ",", "?", keyCols.size() + cols.size());
+    builder.append(")");
+    builder.append(" WITH PRIMARY KEY");
+    log.trace("Upsert Query: " + builder.toString());
```
Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java]: we log this already for each batch, in `BufferedRecords.add()`, so let's remove this log line

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.joinToBuilder;
+import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.nCopiesToBuilder;
+
+public class HANADialect extends DbDialect {
+  private static final Logger log = LoggerFactory.getLogger(HANADialect.class);
```
Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java]: I think we can remove the logger from this class. All of the `DbDialect` methods are of the kind you'd expect the caller to log, as per comments below.

**Comment by shikhar** [comment]
[clabot:check]

@santi81 other than minor logging-related comments this LGTM

**Comment by santi81** [comment]
@shikhar removed the logs.Can you please merge the PR

**Comment by shikhar** [comment]
ok to test

**Pull request merged**